src/modules/pages/components/RouteLayouts/UserNavigation.tsx

```diff
@@ -115,8 +145,177 @@ const UserNavigation = () => {
     }
   }, [userDataLoading, userLock, dispatch, ethereal]);
 
+  type NetworkHealthType = 'healthy' | 'poor' | 'critical';
+  const [networkHealth, setNetworkHealth] = useState<NetworkHealthType>(
+    'healthy',
+  );
+
+  const networkCheckInterval = 30 * 1000; // 15 seconds
+  const serverEndpoint = process.env.SERVER_ENDPOINT;
+  const rpcEndpoint =
+    NETWORK_DATA[process.env.NETWORK || DEFAULT_NETWORK].rpcUrl;
+  const reputationOracleEndpoint =
+    process.env.NODE_ENV === 'development'
+      ? 'http://localhost:3001/reputation'
+      : `${window.location.origin}/reputation`;
+
+  const apolloClient = useApolloClient();
+  const TIMEOUT = 10 * 1000; // Timeout requests after 10 seconds
+  const networkHealthLoadingTime = 1 * 2000; // Show the mini spinner loader for 2 seconds
+
+  const networkNameForReputationOracle =
+    process.env.NODE_ENV === 'development'
+      ? 'local'
+      : NETWORK_DATA[
+          process.env.NETWORK || DEFAULT_NETWORK
+        ].shortName.toLowerCase(); // @TODO have to make sure this will work for other networks than xdai
+
+  useEffect(() => {
+    let latestBlockFromRpc: string;
+
+    const networkCheck = setInterval(async () => {
+      const rpcBlockResponsePromise = raceAgainstTimeout(
+        fetch(rpcEndpoint, {
+          method: 'POST',
+          body: JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'eth_getBlockByNumber',
+            params: ['latest', false],
+            id: 1,
+          }),
+        }),
+        TIMEOUT,
+        new Error('Timeout reached trying to get latest block from RPC'),
+      );
+
+      // Is the server alive?
+      const serverResponsePromise = raceAgainstTimeout(
+        fetch(`${serverEndpoint}/liveness`),
+        TIMEOUT,
+        new Error('Timeout reached trying to reach server'),
+      );
+
+      const reputationOracleResponsePromise = raceAgainstTimeout(
+        fetch(`${reputationOracleEndpoint}/${networkNameForReputationOracle}`),
+        TIMEOUT,
+        new Error('Timeout reached trying to reach reputation oracle.'),
+      );
+
+      const latestBlockFromSubgraphPromise = raceAgainstTimeout(
+        getLatestSubgraphBlock(apolloClient),
+        TIMEOUT,
+        new Error('Timeout reached trying to get latest block from subgraph'),
+      );
+
+      try {
+        const [
+          rpcBlockResponse,
+          latestBlockFromSubgraph,
+          serverResponse,
+          reputationOracleResponse,
+        ] = await Promise.all([
+          rpcBlockResponsePromise,
+          latestBlockFromSubgraphPromise,
+          serverResponsePromise,
+          reputationOracleResponsePromise,
+        ]);
+
+        if (rpcBlockResponse.status !== 200) {
+          setNetworkHealth('critical');
+          return;
+        }
```

Comment:
A lot of services might take more than 10 seconds to respond _(due to other network factors)_, so I think this will trigger even for services that are running.